Assets/Scripts/EnemyController.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyController : MonoBehaviour
+{
+    public float enemySpeed;
+    public int movingRight = 1;
+    public GameObject groundDetector;
+    public float rayDistance;
+    public Animator enemyAnimator;
+    // public PlayerController playerController;
+    // public PlayerHealthController playerHealthController;
+    // public PlayerHealthDisplayController playerHealthDisplayController;
+
+    private void Update()
+    {
+        PatrolEnemy();
+    }
+
+    private void PatrolEnemy()
+    {
+        enemyAnimator.SetBool("IsPatrol", true);
+        transform.Translate(movingRight * Vector2.right * enemySpeed * Time.deltaTime);
+
+        RaycastHit2D hit = Physics2D.Raycast(groundDetector.transform.position, Vector2.down, rayDistance);
+
+        if(!hit)
+        {
+            transform.localScale = new Vector3(-1 * transform.localScale.x, transform.localScale.y, transform.localScale.z);
+            movingRight = movingRight * -1;
+        }
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        if(collision.transform.GetComponent<PlayerController>() != null)
```

Comment:
This can be simplified in a much better way.
`if(collision.transform.TryGetComponent<PlayerController>(PlayerController controller))
{
    controller.DecreaseHealth();
}`

Now what actually is happening here is TryGetComponent works same as GetComponent but it also helps in storing the reference of that particular component in a variable which you can use later.
For example in this case i have controller variable which is of type PlayerController. No need to repeat same code again and again.